DependencyInjection/Configuration.php

```diff
@@ -50,7 +50,12 @@
  *   - [bubble]: bool, defaults to true
  *
  * - gelf:
- *   - publisher: {id: ...} or {hostname: ..., port: ..., chunk_size: ...}
+ *   - publiser:
+ *      - id: string, service id of a publisher implementation, optional if hostname is given
+ *      - hostname: string, optional if id is given
+ *      - [port]: int, defaults to 12201
+ *      - [chunk_size]: int, defaults to 1420
+ *      - [encoder]: string, its value can be 'json' or 'compressed_json'
```

Comment:
Before your change, it was clear that I should either provide a service ID or configure the connection by specifying the hostname, port etc. The host name is not "optional if id is given" as you wrote, it's ignored entirely if a service ID has been configured.

Reply:
```
 * - publisher: (one of the following configurations)
 *      # Option 1: Service-based configuration
 *      - id: string, service id of a publisher implementation
 *      
 *      # Option 2: Direct connection configuration
 *      - hostname: string, server hostname
 *      - [port]: int, server port (default: 12201)
 *      - [chunk_size]: int, UDP packet size (default: 1420)
 *      - [encoder]: string, encoding format ('json' or 'compressed_json')
```

WDYT about this approach?